cookbooks/aws-parallelcluster-shared/attributes/users.rb

```diff
@@ -18,3 +18,19 @@
 default['cluster']['munge']['user_id'] = node['cluster']['reserved_base_uid'] + 2
 default['cluster']['munge']['group'] = node['cluster']['munge']['user']
 default['cluster']['munge']['group_id'] = node['cluster']['munge']['user_id']
+
+if (platform?('amazon') && node['platform_version'].to_i == 2) ||
```

Comment:
because the files weren't adding any value.  Several of them use the same logic and they are all just one line of code

Reply:
The idea of having different files for different users is to simplify addition/removal of an os. Having files with os specific logic will isolate the code for the different OSes.

In past we had a lot of if/else in all the recipes and resources to differentiate the behaviour for the different oses, but was very hard to search for them and fix all of them when adding a new OS.
Having os specific files permits to search for <os-suffix>.rb and remove all of them when deprecating an OS, or create OS specific files if the behaviour is specific for them when adding a new OS.

If we want to reduce the number of files we could create one for "platform" (e.g. unify amazon2 and redhat2) or we can rename users_<os>.rb to a more generic name like environment_<os>.rb to be a container for all the "os" specific attributes we can have in future.

Reply:
I think that makes sense.  The other issue in this case is that we need to know the OS in order to be able to determine the cluster user and then the home directory.  Since the behavior is the same for all the OSes, we end up with the same two lines of code duplicated across 5 different files.  Ruby and Chef don't really allow the kind of hierarchy we need to order our attribute definitions cleanly.  Is there another way?  Would you prefer we have the logic 

```
default['cluster']['cluster_user_home'] = "/home/#{node['cluster']['cluster_user']}"
default['cluster']['cluster_user_local_home'] = "/local#{node['cluster']['cluster_user_home']}"
```
in each OS specific user file?